File: fe-arise/src/dtos/enrollment/EnrollmentDTO.ts
import { AuditInfoDTO } from "../base";
import { StudentDTO } from "../student/StudentDTO";

export type EnrollmentDTO = {
  id: number;
  student: StudentDTO;
  course: number;
  classCode: string;
  className: string;
  courseCode: string;
  courseName: string;
  enrollmentDate: Date;
} & AuditInfoDTO;

File: be-arise/src/main/java/vn/codezx/arise/dtos/enrollment/EnrollmentDTO.java
package vn.codezx.arise.dtos.enrollment;

import java.util.Date;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.SuperBuilder;
import vn.codezx.arise.entities.base.BaseInfo;
import vn.codezx.arise.entities.student.Student;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@SuperBuilder
public class EnrollmentDTO extends BaseInfo {
  private int id;
  private Student student;
  private int courseId;
  private String className;
  private String classCode;
  private String courseCode;
  private String courseName;
  private Date enrollmentDate;
}
